Add support for sdl-only platform

Hello,

In the rlvm fork I use for linux arm64, there are a few commits that could be ported to SuperEngine. These commits add support for SDL-only building, dropping reliance on gtk+ and some other dependencies.

https://github.com/JeodC/rlvm/commit/54184d6c9f821be3da8d6d72bf29de4411941708
https://github.com/JeodC/rlvm/commit/cf76d9d651469abf027b93abaf5ecc8f314d727e

If I can build with `--platform sdl`, then I can give this a try. I haven't tried adding these in myself since I am not familiar with the changes you've made to decoupling platforms.

> Yes, I added a [patch](https://github.com/Mickey-snow/SuperEngine/commit/d7000df88728337966de82159510e4b771d070a4) to remove GTK dependencies and restructure the building system. Now you should be able to compile it as you normally would.

I think that will work, but perhaps something isn't quite right? I'm happy to compile and test current vnovels if you would like. Again thank you for your efforts with this! Unfortunately I don't know much about adding *new* code, so I can't assist with development too well. I typically hack away at existing projects and see what sticks.
```
 cmake -S . -B build -G "Ninja" -DCMAKE_BUILD_TYPE=Release -DRLVM_FULL_STATIC=ON
-- The CXX compiler identification is GNU 10.2.1
-- The C compiler identification is GNU 10.2.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Found OpenGL: /usr/lib/aarch64-linux-gnu/libOpenGL.so
-- OpenGL found:
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Found SDL: /usr/lib/aarch64-linux-gnu/libSDLmain.a;/usr/lib/aarch64-linux-gnu/libSDL.so;-lpthread (found version "1.2.15")
-- SDL1 found: 1.2.15
-- Found Boost: /usr/lib/aarch64-linux-gnu/cmake/Boost-1.74.0/BoostConfig.cmake (found version "1.74.0") found components: program_options serialization iostreams filesystem
-- Boost found: 1.74.0
-- Found Freetype: /usr/lib/aarch64-linux-gnu/libfreetype.so (found version "2.10.4")
-- Freetype2 found:
-- Found ZLIB: /usr/lib/aarch64-linux-gnu/libz.so (found version "1.2.11")
-- zlib found:
CMake Error at CMakeLists.txt:81 (find_package):
  By not providing "FindOgg.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Ogg", but
  CMake did not find one.

  Could not find a package configuration file provided by "Ogg" with any of
  the following names:

    OggConfig.cmake
    ogg-config.cmake

  Add the installation prefix of "Ogg" to CMAKE_PREFIX_PATH or set "Ogg_DIR"
  to a directory containing one of the above files.  If "Ogg" provides a
  separate development package or SDK, be sure it has been installed.


-- Configuring incomplete, errors occurred!
See also "/rlvm/build/CMakeFiles/CMakeOutput.log".
See also "/rlvm/build/CMakeFiles/CMakeError.log".
apt install libogg-dev
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
libogg-dev is already the newest version (1.3.4-0.1).
libogg-dev set to manually installed.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

That works, however I come across a new error when using static libraries:
```
cmake -S . -B build -G "Ninja" -DCMAKE_BUILD_TYPE=Release -DRLVM_FULLSTATIC=ON
CMake Warning at CMakeLists.txt:21 (message):
  Building static targets.  This is currently untested and not guaranteed to
  work.


-- OpenGL found:
-- Checking for modules 'sdl;--static'
--   Must specify package names on the command line
CMake Error at /usr/share/cmake-3.24/Modules/FindPkgConfig.cmake:607 (message):
  A required package was not found
Call Stack (most recent call first):
  /usr/share/cmake-3.24/Modules/FindPkgConfig.cmake:829 (_pkg_check_modules_internal)
  CMakeLists.txt:45 (pkg_check_modules)
  ```
  
  I think I can resolve this bit myself given some time.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,9 +23,7 @@ endif()
 
 find_package(PkgConfig REQUIRED)
 if(RLVM_FULLSTATIC)
-  set(_PKG_CONFIG_ARGS "--static")
-else()
-  set(_PKG_CONFIG_ARGS "")
+  list(APPEND PKG_CONFIG_EXECUTABLE --static)
 endif()
 
 enable_testing()
@@ -66,7 +64,7 @@ endif()
 # -------------------------------------------------------------------
 # Find SDL1 Library
 # -------------------------------------------------------------------
-pkg_check_modules(SDL1 REQUIRED sdl "${_PKG_CONFIG_ARGS}")
+pkg_check_modules(SDL1 REQUIRED sdl)
 include_directories(${SDL1_INCLUDE_DIRS})
 link_directories(${SDL1_LIBRARY_DIRS})
 

